### DISCUSSION TITLE
Timeout issue

### DISCUSSION THREAD

**Discussion by chintan-donda** [encode/httpx#2824] [General] [answered]
I'm developing a Gradio based application. I need to use [`googletrans`](https://pypi.org/project/googletrans/) lib for language translation.
`googletrans` lib internally installs httpx. This library uses `httpx` for HTTP requests so HTTP/2 is supported by default.

My application uses machine learning model inference which takes more than 7-8 seconds sometimes.
After inference, the application sends the response to Gradio front-end which gets showcased in the localhost.

Issue:
Because of default Timeout = 5.0 seconds in httpx, it is causing Gradio application failed to receive the response.

When I change [this line](https://github.com/encode/httpx/blob/master/httpx/_config.py#L367) to 10 seconds, it works fine.

How can we overwrite this value? Is there any fix?

**Comment by karpetrosyan** [comment]
Hi! You don't have access to the httpx client, but you'd like to change the timeout, correct? Could you please provide more information and possibly some code examples so that we can better understand your problem?

**Comment by chintan-donda** [comment]
@karosis88 
I don't want the access to httpx client.
Rather I want to change the timeout something like:
```
import httpx
from httpx._config import Timeout
httpx._config.DEFAULT_TIMEOUT_CONFIG = Timeout(timeout=30.0)
print(f"httpx DEFAULT_TIMEOUT_CONFIG set to: {httpx._config.DEFAULT_TIMEOUT_CONFIG}")
```

This prints on console that DEFAULT_TIMEOUT_CONFIG = 30.0
But this config doesn't get applied globally.

When I run my application, I still get the timeout issue.

**Comment by karpetrosyan** [comment]
I don't think it's a reliable solution; instead, find some api that allows you to change the client's timeout.

In your case, httpx client is used by "googletrans" library, right?

**Comment by chintan-donda** [comment]
> I don't think it's a reliable solution; instead, find some api that allows you to change the client's timeout.
> 
> In your case, httpx client is used by "googletrans" library, right?

yes, "googletrans" library uses httpx client.

I've found something like this from [here](https://www.python-httpx.org/advanced/#setting-and-disabling-timeouts):
```
client = httpx.Client(timeout=10.0)
with httpx.Client(timeout=20.0) as client:
    client.get("http://example.com/api/v1/example", timeout=None)
```

But not working out. For this, I need to make changes inside googletrans lib. Not feasible.

**Comment by karpetrosyan** [comment]
> @karosis88
I don't want the access to httpx client.
Rather I want to change the timeout something like:

If you do this before importing the "googletrans" library, your example should work.

**Comment by karpetrosyan** [comment]
Alternatively, you can import that library, then find the client if it's globally defined and change the timeout property.

**Reply by chintan-donda** [reply]
Tried this, but doesn't work either:
```
from googletrans import Translator
# Init the Google API translator
translator = Translator(timeout=30.0)
```

**Comment by chintan-donda** [comment]
Only solution for now I see is manually update [this line](https://github.com/encode/httpx/blob/master/httpx/_config.py#L367) with higher number.
Any better suggession?

**Reply by karpetrosyan** [reply]
This question is more about "googletrans" than httpx, and I only see that solution if there is no public API to do so.

**Reply by chintan-donda** [reply]
May be yes. But if we can change httpx.DEFAULT_TIMEOUT_CONFIG to configurable value globally, the issue is solved. But couldn't able to do it till now via programatically in Python. Changing manually works.

**Reply by karpetrosyan** [reply, marked as answer]
Try this

``` python
from googletrans import Translator
import httpx
translator = Translator(timeout=httpx.Timeout(0.00000001))

translator.client.get("https://google.com")  # timeout error
```

Worked for me
